app/forms/keywords_form.rb

```diff
@@ -0,0 +1,47 @@
+# frozen_string_literal: true
+
+require 'csv'
+
+class KeywordsForm
+  include ActiveModel::Model
+
+  validates_with KeywordsFormValidator
+
+  attr_accessor :file, :user, :keywords
+
+  def save(params)
+    assign_attributes(params)
+
+    return false if invalid?
+
+    begin
+      if parse_keywords(params[:file])
+        Keyword.create(parse_keywords_from_file(keywords)).map { |keyword| keyword['id'] }
+      end
+    rescue ActiveRecord::ActiveRecordError => e
+      errors.add("Error: #{e}")
+    end
+
+    errors.empty?
+  end
+
+  private
+
+  def parse_keywords_from_file(keyword_records)
+    keyword_records.map { |keyword| add_keyword_record(keyword) }
+  end
```

Comment:
How about this? so that we can remove `add_keyword_record(keyword)`

```suggestion
  def parse_keywords_from_file(keyword_records)
    keyword_records.reject(&:blank?).map do |keyword|
      {
        user_id: user.id,
        keyword: keyword
      }
    end
  end
```

Reply:
Yup I like this, much more simpler 🤔 However Rubocop changed `reject` to `compact_blank` 🙄 Updated here https://github.com/aetheryna/google-scraper-ruby/pull/34/commits/40f76b1ebffdddf0f3a49ba6f8c7d370bf2afcad